portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -14,46 +14,43 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-@WebServlet("Login")
+@WebServlet("login")
 public class LoginServlet extends HttpServlet {
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        UserService userService = UserServiceFactory.getUserService();
-        PrintWriter out = response.getWriter();
-
-        if (!userService.isUserLoggedIn()) {
-            String loginUrl = userService.createLoginURL("/Login");
-            out.println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
-        } else {
-            String logoutUrl = userService.createLogoutURL("/");
-            out.println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
-            out.println("<p>Username:");
-            out.println("<form method=\"POST\" action=\"Login\">");
-            out.println("<input name=\"username\" >");
-            out.println("<br />");
-            out.println("<button>Submit</button>");
-            out.println("</form>");
-        }
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    PrintWriter out = response.getWriter();
+
+    if (!userService.isUserLoggedIn()) {
+      String loginUrl = userService.createLoginURL("/Login");
```

Comment:
I assume this instance of "/login" should be lowercased too =)

Reply:
Although also, similar to my other comment about pulling out repeatedly used strings to private static final... we wouldn't have a chance to accidentally be inconsistent with this name if we made it a constant.  Something like:
private static final String LOGIN_PATH = "/login";

could be referenced here and below in the doPost method.  =)

Reply:
lowercased!